docs/modules/ROOT/pages/how-to-guides/managing-applications/proc-managing_applications.adoc

```diff
@@ -1,85 +1,86 @@
+. When you are satisfied that the fix is ready, wait until the embargo disclosure time before releasing your fix.
 
-. Confirm that the YAML file has been generated under the link:https://github.com/redhat-appstudio/rhtap-cli/tree/main/data/20231003121403[`./rhtap/data` directory]. This file contains the necessary information for importing the application into the new workspace.
+== Releasing the security fix
 
-. Log in to the newly created secure workspace using the `oc` CLI tool.
+.**Prerequisite**
 
-. Apply the generated YAML file to the new workspace by using the following command:
+* You have prepared a fix in a private git mirror and private workspace.
+* You have reviewed the fix with representatives from your security department and select members of your team on a need-to-know basis.
+* The date for lifting the embargo has arrived.
 
-+
-[source,bash]
-----
-oc apply -f <path-to-yaml>
-----
-
-. Within the RHTAP UI, import the impacted component into the same application that the user has just imported. This ensures that the application is fully functional in the secure workspace.
+.**Procedure**
 
+. In the private workspace, use the UI to identify the pullspec of the image bearing the fix, pinned to digest.
+. In your team's normal workspace (not the private workspace), use the CLI to identify a `Snapshot` that you want to use as a basis for your new release. A common choice is "the most recently released Snapshot", which you can find by picking through your history of Releases sorted by creation date to find the relevant Snapshot identifier.
+. Use `oc` to extract the Snapshot that you want to use as a basis for your new release to a local yaml file with `oc get snapshots <that snapshot id> -o yaml > snapshot.yaml`
+. Edit the extracted file to construct a new `Snapshot` that replaces the impacted components pullspec with the pullspec of the image bearing the fix, pinned to digest. You can discard all annotations and labels and should give the Snapshot a new `name`.
+. Create that new Snapshot in the system with `oc apply -f snapshot.yaml`.
+. Review your integration test pipelines to confirm that the `Snapshot` is valid and can be released.
+. Create a `Release` resource referencing the specified `Snapshot`. Follow a similar procedure as outlined above. Export the existing `Release` as a starting point. Modify it to reference the new `Snapshot` and make any necessary adjustments.
+. In your team's normal workspace (not the private workspace), use the UI to watch the progress of the `Release`. When completed, your fix is live and available to your customers.
+. Cherry-pick the commit(s) from your private mirror back to the main Git repository used by your team for normal development.
+
+IMPORTANT: Ensure to apply the patch(es) from your private mirror to the Git repository that your team normally uses for managing development. Failure to do so results in security regressions in subsequent releases from your team.
```

Comment:
We still do not address the need to keep the rest of the components up to date in the private workspace to ensure proper functionality in the broader sense. Should we call this out?

Reply:
I added a note in 361c86b. WDYT?